keyboards/silverbullet44/keymaps/default/keymap.c

```diff
@@ -0,0 +1,213 @@
+  XXXXXXX, XXXXXXX, MU_TOG,  CK_UP,   AU_TOG,  XXXXXXX,                      RGB_SPI, RGB_MOD,  RGB_VAI, RGB_SAI, RGB_HUI, XXXXXXX, \
+  XXXXXXX, XXXXXXX, MU_MOD,  CK_DOWN, XXXXXXX, XXXXXXX,                      RGB_SPD, RGB_RMOD, RGB_VAD, RGB_SAD, RGB_HUD, XXXXXXX, \
+                             XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX \
+  )
+};
+
+#ifdef RGBLIGHT_ENABLE
+// Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
+
+int  RGB_current_mode;
+bool alt_pressed = false;
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case ADJUST:
+            if (record->event.pressed) {
+                layer_on(_ADJUST);
+            } else {
+                layer_off(_ADJUST);
+            }
+            return false;
+            break;
+#ifdef RGBLIGHT_ENABLE
+        case RGBRST:
+            if (record->event.pressed) {
+                eeconfig_update_rgblight_default();
+                rgblight_enable();
+                RGB_current_mode = rgblight_config.mode;
+            }
+            break;
+#endif
+        case KC_SCLN:
+            if (keyboard_report->mods & MOD_BIT(KC_LSFT)) {
+                if (record->event.pressed) {
+                    unregister_code(KC_LSFT);
+                    register_code(keycode);
+                    unregister_code(keycode);
+                    register_code(KC_LSFT);
+                }
+                /*  } else if (keyboard_report->mods & MOD_BIT(KC_RSFT)) {
+                    if (record->event.pressed) {
+                      unregister_code(KC_RSFT);
+                      register_code(kc);
+                      unregister_code(kc);
+                      register_code(KC_RSFT);
+                    } */
+            } else {
+                if (record->event.pressed) {
+                    register_code(KC_LSFT);
+                    register_code(keycode);
+                    unregister_code(keycode);
+                    unregister_code(KC_LSFT);
+                }
+            }
+            return false;
+            break;
+        case KC_00:
+            if (record->event.pressed) {
+                SEND_STRING("00");
+            }
+            return false;
+            break;
+        case S(ALTAB):
+        case ALTAB:
+            if (record->event.pressed) {
+                if (!alt_pressed) {
+                    alt_pressed = true;
+                    register_code(KC_LALT);
+                }
+                if (keycode == S(ALTAB)) {
+                    register_code(KC_LSFT);
+                }
+                register_code(KC_TAB);
+            } else {
+                unregister_code(KC_TAB);
+                if (keycode == S(ALTAB)) {
+                    unregister_code(KC_LSFT);
+                }
+            }
+            return false;
+            break;
```

Comment:
This entire thing looks like it could be replaced with: `A(KC_TAB)` in your keymap (short for `LALT(KC_TAB)`)

Reply:
This is a macro for switching windows.
ALTAB switches windows in normal order, S (ALTAB) reverses.
The firmware is in the current state because it needs to hold down ALT until a key other than ALTAB is pressed (or released).

Reply:
Ah, okay